### PR TITLE
Usp 4307 implement the add lines command

### PR DESCRIPTION
**Pull request by agehrin50**

### CODE REVIEW

**Review by erikgabbard — APPROVED** [review]

File: Constants.cs [hunk 1]
```diff
@@ -4,14 +4,16 @@ public class Constants
 {
     public const char StartToken = '{';
     public const char EndToken = '}';
-    public const string TokenSeparator = ", ";
+    public const string TokenSeparator = ",";
```
Comment (Drise13) [inline Constants.cs:7]: This is a breaking change

Comment (Drise13) [reply Constants.cs:7]: Might want to add a comment that the space is requried

File: Handlers/AddLineCommandHandler.cs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        _printerConnections = configuration.Value;
+    }
+
+    public async Task<StringResponse> Handle(AddLineRequest request, CancellationToken cancellationToken)
+    {
+        var printerId = request.Data.ExtractPrinterId();
+        var ipAddress = request.Data.ExtractAdditionalParameter();
+
+        return _printerConnections.Printers.Any(p => p.PrinterId == printerId)
+            ? new StringResponse(AddLine, printerId, false)
```
Comment (Drise13) [inline Handlers/AddLineCommandHandler.cs:32]: ternary format

File: PrinterConnections.json
Comment (Drise13) [inline PrinterConnections.json:1]: Should be git ignored, not commited

**Review by Drise13 — CHANGES_REQUESTED** [review]

File: Constants.cs (hunk 1 above)
Comment (erikgabbard) [reply Constants.cs:7]: The space should not be required.

File: PrinterConnections.json
Comment (erikgabbard) [reply PrinterConnections.json:1]: Why would you ignore this and not commit it to the repo? It is a required file for things to work.

File: Constants.cs (hunk 1 above)
Comment (erikgabbard) [reply Constants.cs:7]: I guess that's fair enough.

File: PrinterConnections.json
Comment (erikgabbard) [reply PrinterConnections.json:1]: It's true that we don't want our own printers, but you can't exclude the file altoghther.

Comment (Drise13) [reply PrinterConnections.json:1]: @agehrin50 This conversation is not resolved. _Something_ should be done about including development settings in the repository, but I'm not 100% clear on the correct course of action. Does this file auto-generate? @Gishe How do we handle this correctly?

Comment (erikgabbard) [reply PrinterConnections.json:1]: @Drise13 What is your actual objection here?  There is nothing sensitive about including a printer name and IP address from our development environment in the repository as an example of what would appear in the file.  The only thing you don't want to commit to a repo is secret info like username and password.

File: BoxwriterResmarkInterop/Handlers/AddLineCommandHandler.cs [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    {
+        _printerConnections = configuration.Value;
+    }
+
+    public async Task<StringResponse> Handle(AddLineRequest request, CancellationToken cancellationToken)
+    {
+        var printerId = request.Data.ExtractPrinterId();
+        var ipAddress = request.Data.ExtractAdditionalParameter();
+
+        return _printerConnections.Printers.Any(p => p.PrinterId == printerId)
```
Comment (Gishe) [inline BoxwriterResmarkInterop/Handlers/AddLineCommandHandler.cs:31]: This does not allow users to update the value if it changes.  I would expect calling it with an existing value would update the value not give me a false back.

**Review by Gishe — CHANGES_REQUESTED** [review]
Please add the example file to the Zip output from the Git action.

File: Configuration/PrinterConnectionInfo.cs [hunk 4]
```diff
@@ -1,8 +1,8 @@
...
-public class PrinterConnection
-{
-    public string? PrinterId { get; set; }
-
-    public string? IpAddress { get; set; }
-}
\ No newline at end of file
+﻿namespace BoxwriterResmarkInterop.Configuration;
+
+public class PrinterConnectionInfo
```
[the remark this answers is not in the document]
Comment (agehrin50) [reply Configuration/PrinterConnectionInfo.cs]: We now have a class that includes a list of `PrinterConnectionInfo` called `PrinterConnections`

File: BoxwriterResmarkInterop/Handlers/AddLineCommandHandler.cs (hunk 3 above)
Comment (agehrin50) [reply BoxwriterResmarkInterop/Handlers/AddLineCommandHandler.cs:31]: I can update that if that is the expectation. My expectation was that calling it with an existing value would not modify the list of connections

Comment (Drise13) [reply BoxwriterResmarkInterop/Handlers/AddLineCommandHandler.cs:31]: I would expect it to replace it with new IP data.

**Review by Drise13 — APPROVED** [review]

**Pull request merged**